client/packages/electron/src/electron.ts

```diff
@@ -119,14 +110,22 @@ app.on('window-all-closed', () => {
 
 app.on(
   'certificate-error',
-  (event, _webContents, url, _error, _certificate, callback) => {
-    if (!connectedServer) return callback(false);
-
-    if (url.startsWith(frontEndHostUrl(connectedServer))) {
-      event.preventDefault();
-      callback(true);
-    } else {
-      callback(false);
-    }
+  (event, _webContents, _url, _error, _certificate, callback) => {
+    event.preventDefault();
+    return callback(true);
+
+    // TODO store an object with this shape: { [hardware_id + port]: cert }, retreive this object on startup
```

Comment:
Added to cary over

Reply:
seems wrong to me, you basically say you accept all connections even if there is something wrong with the cert, i.e. an attacker can pretend to be the remote server using any cert.

What was the idea for the previous code? think just validating the ip is not enough, is it? On the first connection to the remove server we need to fetch the server's public cert and save it (and assume there is no man in the middle at this very first connection, or use another secure channel such as qr code, mail...). For later connections we manually validate that the request cert matches the one we stored and continue the connection. (think it was this point: https://github.com/openmsupply/open-msupply/pull/40#discussion_r882213131)

For Android I do something very similar here:
`packages/android/app/src/main/java/org/openmsupply/client/certplugin/CertPlugin.java`
If there is an error I check if the request cert matches the cert of the local server and if it does I continue with the connection.

Reply:
I've added this comment to the carry over, I think what you propose sounds the safest, but as always with security it's a balance between convenience vs security. 

In my view it's ok to assume that initial connection is secure, I think it's safe to say that client/server discovery is for local network use only, if there is an attacker in your local network, and they've built a malicious version of omSupply in an effort to get users's omSupply password, when the laptop is set up user should see multiple available servers, at which point should notify the admin ? 

I do like what you've suggested though, would be great to implement it, but would suggest it should be done in distant future.